tests/pytorch/test_numerics.py

```diff
@@ -1216,6 +1216,8 @@ def test_layernorm_mlp_accuracy(dtype, bs, model, activation, normalization):
 
 def _test_grouped_linear_accuracy(block, num_gemms, bs, dtype, config, fp8=False):
     reset_rng_states()
+    # using this seed because it will generate cases where m = 0, while the default seed 1234 won't
+    torch.manual_seed(1235)
```

Comment:
This seems delicate. It would be better to manipulate the data directly to avoid this case, perhaps:
```python
    m = config.seq_len // 16
    m_splits = torch.randint(1, m - num_gemms - 1, num_gemms)
    m_splits = torch.msort(m_splits)
    m_splits += torch.arange(m_splits.numel(), dtype=m_splits.dtype)
    m_splits = torch.diff(
        m_splits,
        prepend=torch.tensor([0]),
        append=torch.tensor([m]),
    )
    m_splits = m_splits * 16
    assert m_splits.sum() == config.seq_len and len(m_splits) == num_gemms
```

Reply:
I intended to have zero in `m_splits` for testing.

Reply:
Manually insert a zero now.